cmd/cli/kubectl-kyverno/apply/apply_command.go

```diff
@@ -214,7 +222,40 @@ func (c *ApplyCommandConfig) applyCommandHelper() (rc *common.ResultCounts, reso
 		return rc, resources, skipInvalidPolicies, pvInfos, sanitizederror.NewWithError("a stdin pipe can be used for either policies or resources, not both", err)
 	}
 
-	policies, err := common.GetPoliciesFromPaths(fs, c.PolicyPaths, false, "")
+	isGit := common.IsGitPath(c.PolicyPaths)
+	var policies []kyvernov1.PolicyInterface
+	gitURL, err := url.Parse(c.PolicyPaths[0])
+	if err != nil {
+		fmt.Printf("Error: failed to load policies\nCause: %s\n", err)
+		os.Exit(1)
+	}
+
+	pathElems := strings.Split(gitURL.Path[1:], "/")
+	if len(pathElems) <= 1 {
+		err := fmt.Errorf("invalid URL path %s - expected https://github.com/:owner/:repository/:branch (without --git-branch flag) OR https://github.com/:owner/:repository/:directory (with --git-branch flag)", gitURL.Path)
```

Comment:
GitHub is not the only source of git repositories. Any functionality that works with git repositories should not assume github.com anywhere.

Reply:
The main priority was to add support for git URLs, so I prefer that first. 
Yeah we can add functionality to deal with other sources too, such GitLab, BitBucket.  Any others sources according to you.

Reply:
That's my point, git URLs can come from more than just GitHub. GitHub is not the only git-as-a-Service provider.

Reply:
It supports all URLs having prefix: `https://`.

Reply:
But why is the error message specifically returning `https://github.com/` in the string?

Reply:
When a parsing error occurs, you're printing that line which appears to the user that only git repositories from GitHub are supported, but that isn't the case so the returned message is misleading.

Reply:
> But why is the error message specifically returning `https://github.com/` in the string?
Will this message be fine, `https://<any_git_source_domain>/:owner/:repository/:branch` or something else ?

Reply:
Sure, something like that is fine. Same with my other comment. We should never assume one specific vendor of a service in generic messages we return.

Reply:
agreee